### ISSUE TITLE
Missing .merge and .throttle

### ISSUE THREAD

**Issue by doronkatz** [broadwaylamb/OpenCombine#141]
Hi folks,

So I have the following:
```swift
public func mergedMovies(ids movieIDs: [String]) -> AnyPublisher<[Movie], APIError>{
        let movieIDs = Array(movieIDs.prefix(20))
        
        precondition(!movieIDs.isEmpty)

        let initialPublisher = movie(id: movieIDs[0])
        let remainder = Array(movieIDs.dropFirst())

        #if canImport(Combine)
        return remainder.reduce(initialPublisher) { combined, id in
          return combined
            .merge(with: movie(id: id))
            .eraseToAnyPublisher()
        }
        #else
        return remainder.reduce(initialPublisher) { combined, id in
          return combined
            .merge(with: movie(id: id))
            .eraseToAnyPublisher()
        }
        #endif

    }
```
I am getting two errors:

```swift
value of type 'Any' has no member 'eraseToAnyPublisher'
            .eraseToAnyPublisher()
```

and, 
```swift
value of type 'AnyPublisher<[Movie], APIError>' has no member 'merge'
            .merge(with: movie(id: id))
```

Not sure where `ocombine` should reside to resolve that, if that is the fix.

Thanks

**Comment by broadwaylamb** [comment]
Hi!

The reason is that `merge` and `throttle` are not implemented yet. `merge` is being worked on in #72.

**Comment by doronkatz** [comment]
Got it thanks

**Comment by andybest** [comment]
Hi, I wondered if there has been any progress made on this?

Thanks :)

**Comment by broadwaylamb** [comment]
> Hi, I wondered if there has been any progress made on this?
> 
> 
> 
> Thanks :)

Hi. Nope, I’ve been really busy lately.

**Comment by maximkrouk** [comment]
Hi, I also wondered if there has been any progress made on this?

I'd like to implement [ComposableArchitecture](https://github.com/MakeupStudio/opencombine-composable-architecture) using OpenCombine, but is probably impossible without merge publishers 🙂

Thanks :)

**Comment by broadwaylamb** [comment]
Nope, no progress. I hope I'll find some time in the coming months, but no promises.

**Comment by maximkrouk** [comment]
Btw it is implemented in CombineX, maybe this may help
[Merge](https://github.com/cx-org/CombineX/blob/43037a7ae90e8aa022e478706b854d9a7cea5a70/Sources/CombineX/Publishers/B/Merge.swift)
[MergeMany](https://github.com/cx-org/CombineX/blob/43037a7ae90e8aa022e478706b854d9a7cea5a70/Sources/CombineX/Publishers/B/Combined/Merge%2B.swift)

**Comment by broadwaylamb** [comment]
Unfortunately, the implementation is not that easy, there are many corner cases. I am pretty certain that the one that you refer to doesn’t handle them all, and is not tested very well.

**Comment by maximkrouk** [comment]
Any updates? 🙂

**Comment by broadwaylamb** [comment]
Nope. Doing my best, sorry.

**Comment by Degtyarserg** [comment]
Hi, I wondered if there has been any progress made on this?

Thanks 😉

**Comment by MaxDesiatov** [comment]
No updates on this, but all contributions that help us in the implementation are very welcome.
